toolkit/scripts/toolchain/download_toolchain_rpm.sh

```diff
@@ -139,6 +158,32 @@ function download() {
     exit 1
 }
 
+function validate_signatures() {
+    if [ -z "$work_dir" ]; then
+        work_dir=$(mktemp -d)
+        function cleanup() {
+            rm -rf "$work_dir"
+        }
+        trap cleanup EXIT
+    fi
```

Comment:
I understand this is the only place we're using a work directory right now, but I'd still suggest moving the set-up and clean-up to the global prep before we run any other logic. Bash scripts allow you to set-up only one function call for a trap, so it's good to expose all traps in a visible spot, so future changes can append to it. Otherwise, it's easy to accidentally try to set-up a separate trap elsewhere for the same signal and run into problems.